solidity/dashboard/src/reducers/modal.js

```diff
@@ -0,0 +1,35 @@
+const initialState = {
+  isOpen: false,
+  emittedEvent: null,
+  transactionHash: null,
+  additionalData: null,
+}
+
+const modalReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case "modal/is_opened":
+      return {
+        ...state,
+        isOpen: true,
+        emittedEvent: action.payload.emittedEvent || state.emittedEvent ,
+        transactionHash: action.payload.transactionHash,
+        additionalData: action.payload.additionalData,
+      }
+    case "modal/is_closed":
+      return {
+        ...state,
+        isOpen: false,
+        emittedEvent: null,
+        transactionHash: null,
+      }
+    case "modal/set_emitted_event":
+      return {
+        ...state,
+        emittedEvent: action.payload,
+      }
+    default:
+      return state
+  }
+}
+
+export default modalReducer
```

Comment:
I think this reducer should look the same as message reducer. So we should have only `modal/open` and `modal/close`. `emittedEvent` and `transactionHash` are not common for all modal components so we should pass them in eg. `modalProps` or `additionalData` as we display the [messages](https://github.com/keep-network/keep-core/blob/main/solidity/dashboard/src/sagas/staking.js#L276-L284).

Reply:
Fixed in cd9abad98ab755f57b1cd5ff39a2afce4545fe20